FirebaseRemoteConfig/Sources/RCNFetch.m

```diff
@@ -215,14 +215,27 @@ - (void)refreshInstanceIDTokenAndFetchCheckInInfoWithCompletionHandler:
                     @"Failed to register InstanceID with error : %@.", error);
       }
       if (token) {
-        NSError *IIDError;
         strongSelf->_settings.configInstanceIDToken = [token copy];
-        strongSelf->_settings.configInstanceID = [instanceID appInstanceID:&IIDError];
-        FIRLogInfo(kFIRLoggerRemoteConfig, @"I-RCN000022", @"Success to get iid : %@.",
-                   strongSelf->_settings.configInstanceID);
+        __weak RCNConfigFetch *instanceIDSelf = strongSelf;
+        [instanceID getIDWithHandler:^(NSString *_Nullable identity, NSError *_Nullable error) {
+          RCNConfigFetch *instanceIDStrongSelf = instanceIDSelf;
+          instanceIDStrongSelf->_settings.configInstanceID = identity;
```

Comment:
Should `_settings` be modified from the `_lockQueue`? `[instanceID getIDWithHandler:]` is called on the main queue. I wonder if the code will be simpler if we call `[instanceID getIDWithHandler:]` first (before line 209) and move the code with `dispatch_async(instanceIDHandlerSelf->_lockQueue` inside the `getIDWithHandler:` handler?

Reply:
Good pt. Done.